db/migrate/20220513214928_convert_database_to_utf8_mb4.rb

```diff
@@ -0,0 +1,63 @@
+class ConvertDatabaseToUtf8Mb4 < ActiveRecord::Migration
+  def up
+    connection.execute "ALTER DATABASE #{ActiveRecord::Base.connection.current_database} CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_520_ci';"
```

Comment:
We don't need to specifify the database name in the query:
```suggestion
    connection.execute "ALTER DATABASE CURRENT CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_520_ci';"
```

Reply:
`CHARACTER SET 'utf8mb4'` is technically unnecessary because it's implied by the collation. But it doesn't hurt to make it explicit.

Reply:
The `CURRENT` trick didn't work. Maybe it's for a newer MySQL version?

```
docker-compose run --rm web rake db:migrate
Creating cdx_web_run ... done
== 20220513214928 ConvertDatabaseToUtf8Mb4: migrating =========================
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

Mysql2::Error: Can't create/write to file './CURRENT/db.opt' (Errcode: 2 - No such file or directory): ALTER DATABASE CURRENT CHARACTER SET 'utf8mb4' COLLATE 'utf8mb4_unicode_520_ci';/usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `_query'
/usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `block in query'
/usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `handle_interrupt'
/usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `query'
/usr/local/bundle/gems/activerecord-4.2.11.3/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:317:in `block in execute'
/usr/local/bundle/gems/activerecord-4.2.11.3/lib/active_record/connection_adapters/abstract_adapter.rb:484:in `block in log'
/usr/local/bundle/gems/activesupport-4.2.11.3/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/usr/local/bundle/gems/activerecord-4.2.11.3/lib/active_record/connection_adapters/abstract_adapter.rb:478:in `log'
/usr/local/bundle/gems/activerecord-4.2.11.3/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:317:in `execute'
/usr/local/bundle/gems/activerecord-4.2.11.3/lib/active_record/connection_adapters/mysql2_adapter.rb:217:in `execute'
/src/db/migrate/20220513214928_convert_database_to_utf8_mb4.rb:3:in `up'
[ ... ]
```

Reply:
Lol so [`CURRENT`](https://stackoverflow.com/a/27043201/641451) is for SQL Server, but in MySQL you can [just skip the DB name](https://stackoverflow.com/a/71981679/641451).

Reply:
Oops, sorry! 🙈